routers/web/repo/issue.go

```diff
@@ -1636,6 +1638,27 @@ func ViewIssue(ctx *context.Context) {
 	ctx.Data["HasProjectsWritePermission"] = ctx.Repo.CanWrite(unit.TypeProjects)
 	ctx.Data["IsRepoAdmin"] = ctx.IsSigned && (ctx.Repo.IsAdmin() || ctx.User.IsAdmin)
 	ctx.Data["LockReasons"] = setting.Repository.Issue.LockReasons
+	var hiddenEvents *forms.UpdateCommentTypeForm
+	if ctx.IsSigned {
+		eventsRaw, err := user_model.GetSettings(ctx.User.ID, []string{"hidden_comment_types"})
```

Comment:
I think we need a rule about the setting key name. @go-gitea/maintainers

Reply:
> I think we need a rule about the setting key name.

I assume you mean a rule about how to name such key name.

I guess it can be very simple:
- Lowercase,
- Use `_` as "space" between words.
- Don't have a reference that the setting is for a specific user(e.g. `user_hidden_comment_types`) as it's already stored on the user's ID itself, so it doesn't make any sense to use it.
- Optional: Always assume that developers should be able to read it in a way like `User's xxx` e.g. `User's hidden comment types` in this scenario.

Reply:
I vote for the same structure like our translation keys `abc.xyz.dummy`.
In my package PR I have something similiar and use `<package type>.<property name>`, example: `npm.tag`

Reply:
I like the idea of using structure like in translations names, for current one adding prefix `issue.` would be appropriate

Reply:
Or maybe `issue.comments.hidden_types`?